sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/AstyanaxDataWriterDAO.java

```diff
@@ -277,7 +288,10 @@ public void compact(Table tbl, String key, UUID compactionKey, Compaction compac
             // Should synchronously write compaction and then delete deltas
             writeCompaction(rowKey, compactionKey, compaction, consistency, placement, keyspace, tbl, key);
 
-            deleteCompactedDeltas(rowKey, consistency, placement, keyspace, changesToDelete, historyList, tbl, key);
+            DateTime cutoffTimeForDeletes = CompactionControlUtils.getOldestStashStartTime(_defaultCompactionControlManager, _dataCenters);
+            if (cutoffTimeForDeletes == null || cutoffTimeForDeletes.getMillis() > System.currentTimeMillis()) {
```

Comment:
I think this check makes more sense here in https://github.com/bazaarvoice/emodb/blob/master/sor/src/main/java/com/bazaarvoice/emodb/sor/core/DistributedCompactor.java#L52